stdlib/map.mli

```diff
@@ -223,7 +223,25 @@ module type S =
     (** Same as {!Map.S.map}, but the function receives as arguments both the
        key and the associated value for each binding of the map. *)
 
+    (** {6 Iterators} *)
 
+    type 'a cursor
+    (** Cursor to iterate over bindings
+        @since NEXT_RELEASE *)
+
+    val cursor_start : 'a t -> 'a cursor
+    (** Iterate on values, in ascending order of their corresponding key
+        @since NEXT_RELEASE *)
+
+    val cursor_start_at : key -> 'a t -> 'a cursor
+    (** [cursor_start_at k m] iterates on a subset of the bindings of [m],
+        in ascending order, from key [k] or above.
+        @since NEXT_RELEASE *)
+
```

Comment:
Not sure adding ad-hoc cursors to the stdlib is the right path to take. But in any case I would simply call the cursor creating function `cursor` and `cursor_at`. Also the documentation of the current `cursor_start` should follow the `[cursor_start m] is` style.